File: Top 100 Codes/01-Getting Started/11-Sum of digits of a number/Sumofdigitsofanumber.py
def find_sum(num):
    """
    Recursively calculates the sum of the digits of a given number.
    
    Args:
        num (int): The number whose digits are to be summed up.
    
    Returns:
        int: The sum of the digits.
    """
    if num == 0:
        return 0
    # Add the last digit to the sum of the remaining digits
    return num % 10 + find_sum(num // 10)


# Get user input
num = int(input("Enter a number: "))

# Validate that the number is non-negative
if num < 0:
    print("Please enter a non-negative number.")
else:
    # Calculate and display the sum of digits
    print(f"The sum of the digits of {num} is: {find_sum(num)}")

File: Top 100 Codes/01-Getting Started/11-Sum of digits of a number/Sumofdigitsofanumber.cpp
#include<iostream>
using namespace std;

// Function to calculate the sum of digits of a number using recursion
int getSum(int num) {
    // Base case: If the number becomes 0, return 0
    if (num == 0)
        return 0;
    // Recursive case: Add the last digit to the sum of the rest of the digits
    return (num % 10) + getSum(num / 10);
}

int main() {
    int num;

    // Prompt the user for input
    cout << "Enter a number: ";
    cin >> num;

    // Ensure the number is positive for simplicity
    if (num < 0) {
        cout << "Please enter a positive number." << endl;
        return 1;
    }

    // Calculate the sum of digits
    int sum = getSum(num);

    // Display the result
    cout << "The sum of the digits of " << num << " is: " << sum << endl;

    return 0;
}

// Time complexity: O(N), where N is the number of digits in the number
// Space complexity: O(N), due to the recursion call stack
